inc/Engine/Admin/RocketInsights/Subscriber.php

```diff
@@ -552,4 +553,36 @@ public function on_update_cancel_old_as_jobs( $new_version, $old_version ) {
 		}
 		$this->queue->deprecate_old_actions();
 	}
+
+	/**
+	 * Callback for the wp_rocket_upgrade action to add homepage on version update if no pages exist.
+	 *
+	 * This method adds the homepage to Rocket Insights when:
+	 * - Updating from a version lower than 3.20.0.2
+	 * - Updating to version 3.20.1 or higher
+	 * - No pages are currently added to Rocket Insights
+	 *
+	 * @param string $new_version New plugin version.
+	 * @param string $old_version Previous plugin version.
+	 * @return void
+	 */
+	public function on_update_add_homepage( $new_version, $old_version ) {
+		// Only run if updating from a version lower than 3.20.0.2.
+		if ( version_compare( $old_version, '3.20.0.2', '>=' ) ) {
+			return;
+		}
+
+		// Only run if updating to 3.20.1 or higher.
+		if ( version_compare( $new_version, '3.20.1', '<' ) ) {
+			return;
+		}
```

Comment:
I believe this condition is no longer necessary: https://group-onecom.slack.com/archives/C08EFCYRQ2X/p1761726076690829?thread_ts=1761720344.111189&cid=C08EFCYRQ2X